### DISCUSSION TITLE
AvaloniaUI Android Project JSON Config File Null Reference issue.

### DISCUSSION THREAD

**Discussion by RedEye-Developers** [AvaloniaUI/Avalonia#17433] [Q&A]
i have null reference issue in AvaloniaUI Android project in AppSettings.json File if i run the code in Desktop the code is running Properly  and i get correct result if i run the same code in android device the application was crash and i debug the application some json file related function throwing null reference error.

private static readonly string appSettingsFilePath = "Configs\AppSettings.json";
if i use the path like this still same issue

i am using both this package in both Main and Android Project

microsoft.extensions.configuration
microsoft.extensions.configuration.json

```cs
UpdateService.cs

public async Task<GitAppConfig> GetGitAppConfigAsync()
    {
        string gitURL = AppSettings.GetAppSettings("AppSettings:GitURL");
        using (HttpClient client = new HttpClient())
        {
            try
            {
                string jsonString = await client.GetStringAsync(gitURL);
                GitAppConfig gitAppConfig = JsonConvert.DeserializeObject<GitAppConfig>(jsonString);
                return gitAppConfig;
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }
    }

    public async Task<bool> IsUpdateAvailableAsync()
    {
        string version = AppSettings.GetAppSettings("AppSettings:Version");
        GitAppConfig gitAppConfig = await GetGitAppConfigAsync();
        return version != gitAppConfig.Version;
    }
```

```cs
AppSettings.cs

private static readonly string appSettingsFilePath = @"D:\Others\Software_Programming_Scripts\FitBites\FitBites\FitBites\Configs\AppSettings.json";
    public static IConfiguration Configuration { get; private set; }
    
    static AppSettings()
    {
        var builder = new ConfigurationBuilder()
            .SetBasePath(Directory.GetCurrentDirectory())
            .AddJsonFile(appSettingsFilePath, optional:false, reloadOnChange:true);
        
        Configuration = builder.Build();
    }

    public static string GetAppSettings(string key)
    {
        return Configuration[key];
    }
```

![Screenshot (22)](https://github.com/user-attachments/assets/c836e63e-e78b-4ba5-94df-b13e57eb97b9)
![Screenshot (23)](https://github.com/user-attachments/assets/ab5f8866-1ff0-4945-934b-bee6ad0167f4)

**Comment by thevortexcloud** [comment]
Just a couple things to check:

> private static readonly string appSettingsFilePath = @"D:\Others\Software_Programming_Scripts\FitBites\FitBites\FitBites\Configs\AppSettings.json";

This is a Windows path and wont work on Android. I see you show a different path in the discussion text but make sure you actually have it in the code.

Second, make sure you actually have permission to access the file system. Android has a fairly complicated permission system that requires fiddling with manifest files and then prompting for, or manually granting access. Some directories however do not require permission to access, but they are typically only accessible to the app itself and nothing else.

https://developer.android.com/guide/topics/permissions/overview

https://developer.android.com/reference/android/Manifest.permission
